src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlDictionaryAsyncCheckWriter.cs

```diff
@@ -11,17 +11,17 @@
 
 namespace System.Xml
 {
-    internal sealed class XmlDictionaryAsyncCheckWriter : XmlDictionaryWriter
+    internal sealed class XmlDictionaryAsyncCheckWriter<T> : XmlDictionaryWriter, IXmlTextWriterInitializer where T : XmlDictionaryWriter, IXmlTextWriterInitializer
```

Comment:
Why does this need to be generic rather than just hardcoding the T to be XmlUTF8TextWriter?

Reply:
Right now, `XmlUTF8TextWriter` is the only implementer of the interface. But perhaps there are other text writers in the future that might also implement the interface? I don't really see that as a likely scenario, but the thought did cross my mind and this just felt like the cleaner approach in that case.

But as I said in the PR description, this is all internal and I did initially start in the other direction. If there's good reason to not use the generic approach, I don't mind switching it up.

Reply:
It's just unnecessary expense.  If in the future a generic is needed, it can be changed then.